From: Mitsutoshi NAKANO <ItSANgo@gmail.com>
Date: Sat, 20 Jun 2015 18:36:55 +0900
Subject: fix-print_msg_getc

modify print_msg_getc(): use vsnprintf() and ANSI style.

	modified:   ChangeLog
	modified:   ChangeLog.en
	modified:   Wnn/uum/sdefine.h
	modified:   Xwnmo/xjutil/uif.c
	modified:   Xwnmo/xjutil/xext.h
	modified:   Xwnmo/xwnmo/ext.h
	modified:   Xwnmo/xwnmo/printf.c
---
 ChangeLog            | 25 +++++++++++++++++++++++++
 ChangeLog.en         | 25 +++++++++++++++++++++++++
 Wnn/uum/sdefine.h    |  4 ++--
 Xwnmo/xjutil/uif.c   |  4 +++-
 Xwnmo/xjutil/xext.h  |  4 +++-
 Xwnmo/xwnmo/ext.h    |  4 +++-
 Xwnmo/xwnmo/printf.c | 15 +++++++++++----
 7 files changed, 72 insertions(+), 9 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 9d1a25c..b633087 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -3,6 +3,31 @@
   Japanese...
  */
 
+Sat, 20 Jun 2015 17:56:56  Mitsutoshi NAKANO  <ItSANgo@gmail.com>
+
+	* ChangeLog.en: added these messages.
+	* ChangeLog: added these messages.
+
+	* Xwnmo/xwnmo/printf.c: added Copyright FreeWnn Project 2015 in comment.
+	* Xwnmo/xwnmo/printf.c: added Maintainer: FreeWnn Project in comment.
+	* Xwnmo/xwnmo/printf.c: include <stdarg.h> for print_msg_getc(format, ...).
+	* Xwnmo/xwnmo/printf.c (print_msg_getc): fixed to ANSI style, and used vsnprintf().
+	* Xwnmo/xwnmo/ext.h: added Copyright FreeWnn Project 2015 in comment.
+	added Maintainer: FreeWnn Project.
+	fixed print_msg_getc() to ANSI style.
+
+	* Xwnmo/xjutil/xext.h: fixed print_msg_getc() to ANSI style.
+	added Maintainer: FreeWnn Project.
+	fixed print_msg_getc() to ANSI style.
+
+	* Xwnmo/xjutil/uif.c (isconect_jserver): removed NULL arguments.
+	* Xwnmo/xjutil/uif.c: added Copyright FreeWnn Project 2015.
+	added  Maintainer:  FreeWnn Project.
+
+	* Wnn/uum/sdefine.h (print_msg_getc)
+	(print_msg): fix printf(X) -> printf("%s", (X)) in macros.
+	* Wnn/uum/sdefine.h: added Copyright 2015.
+
 Sun, 14 Jun 2015 06:41:18  Mitsutoshi NAKANO  <ItSANgo@gmail.com>
 
 	* Wnn/uum/kensaku.c(word_list1): fixed printf format.
diff --git a/ChangeLog.en b/ChangeLog.en
index 1f4e941..db934ae 100644
--- a/ChangeLog.en
+++ b/ChangeLog.en
@@ -1,3 +1,28 @@
+Sat, 20 Jun 2015 17:56:56  Mitsutoshi NAKANO  <ItSANgo@gmail.com>
+
+	* ChangeLog.en: added these messages.
+	* ChangeLog: added thse messages.
+
+	* Xwnmo/xwnmo/printf.c: added Copyright FreeWnn Project 2015 in comment.
+	* Xwnmo/xwnmo/printf.c: added Maintainer: FreeWnn Project in comment.
+	* Xwnmo/xwnmo/printf.c: include <stdarg.h> for print_msg_getc(format, ...).
+	* Xwnmo/xwnmo/printf.c (print_msg_getc): fixed to ANSI style, and used vsnprintf().
+	* Xwnmo/xwnmo/ext.h: added Copyright FreeWnn Project 2015 in comment.
+	added Maintainer: FreeWnn Project.
+	fixed print_msg_getc() to ANSI style.
+
+	* Xwnmo/xjutil/xext.h: fixed print_msg_getc() to ANSI style.
+	added Maintainer: FreeWnn Project.
+	fixed print_msg_getc() to ANSI style.
+
+	* Xwnmo/xjutil/uif.c (isconect_jserver): removed NULL arguments.
+	* Xwnmo/xjutil/uif.c: added Copyright FreeWnn Project 2015.
+	added  Maintainer:  FreeWnn Project.
+
+	* Wnn/uum/sdefine.h (print_msg_getc)
+	(print_msg): fix printf(X) -> printf("%s", (X)) in macros.
+	* Wnn/uum/sdefine.h: added Copyright 2015.
+
 Sun, 14 Jun 2015 06:41:18  Mitsutoshi NAKANO  <ItSANgo@gmail.com>
 
 	* Wnn/uum/kensaku.c(word_list1): fixed printf format.
diff --git a/Wnn/uum/sdefine.h b/Wnn/uum/sdefine.h
index 305c4cc..8a63850 100644
--- a/Wnn/uum/sdefine.h
+++ b/Wnn/uum/sdefine.h
@@ -191,8 +191,8 @@ struct kansuu
 
 
 #define flush() fflush(stdout)
-#define print_msg(X) {push_cursor();throw_c(0); clr_line();printf(X);flush();pop_cursor();}
-#define print_msg_getc(X) {push_cursor();throw_c(0); clr_line();printf(X);flush();keyin();pop_cursor();}
+#define print_msg(X) {push_cursor();throw_c(0); clr_line();printf("%s", (X));flush();pop_cursor();}
+#define print_msg_getc(X) {push_cursor();throw_c(0); clr_line();printf("%s", (X));flush();keyin();pop_cursor();}
 
 
 #define UNDER_LINE_MODE (0x02 | 0x08 | 0x20)
diff --git a/Xwnmo/xjutil/uif.c b/Xwnmo/xjutil/uif.c
index 7131826..c38c45a 100644
--- a/Xwnmo/xjutil/uif.c
+++ b/Xwnmo/xjutil/uif.c
@@ -11,8 +11,10 @@
  * Copyright OMRON Corporation. 1987, 1988, 1989, 1990, 1991, 1992, 1999
  * Copyright ASTEC, Inc. 1987, 1988, 1989, 1990, 1991, 1992
  * Copyright 1991, 1992 by Massachusetts Institute of Technology
+ * Copyright FreeWnn Project 2015
  *
  * Author: OMRON SOFTWARE Co., Ltd. <freewnn@rd.kyoto.omronsoft.co.jp>
+ * Maintainer:  FreeWnn Project
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -113,7 +115,7 @@ isconect_jserver ()
     }
   if (js_isconnect (cur_env))
     {
-      print_msg_getc (" jserver(?)", NULL, NULL, NULL);
+      print_msg_getc (" jserver(?)");
       t_print_l ();
       return (0);
     }
diff --git a/Xwnmo/xjutil/xext.h b/Xwnmo/xjutil/xext.h
index 7f1fd8f..74f313d 100644
--- a/Xwnmo/xjutil/xext.h
+++ b/Xwnmo/xjutil/xext.h
@@ -10,8 +10,10 @@
  * Copyright OMRON Corporation. 1987, 1988, 1989, 1990, 1991, 1992, 1999
  * Copyright ASTEC, Inc. 1987, 1988, 1989, 1990, 1991, 1992
  * Copyright 1991, 1992 by Massachusetts Institute of Technology
+ * Copyright FreeWnn Project.  2015
  *
  * Author: OMRON SOFTWARE Co., Ltd. <freewnn@rd.kyoto.omronsoft.co.jp>
+ * Maintainer:  FreeWnn Project
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -287,7 +289,7 @@ extern void print_out3 ();
 extern void print_out2 ();
 extern void print_out1 ();
 extern void print_out ();
-extern void print_msg_getc ();
+extern void print_msg_getc (register char *format, ...);
 extern void print_msg_wait ();
 
 /*
diff --git a/Xwnmo/xwnmo/ext.h b/Xwnmo/xwnmo/ext.h
index 9b788f5..d49a68b 100644
--- a/Xwnmo/xwnmo/ext.h
+++ b/Xwnmo/xwnmo/ext.h
@@ -8,8 +8,10 @@
  * 
  * Copyright OMRON Corporation. 1987, 1988, 1989, 1990, 1991, 1992, 1999
  * Copyright 1991, 1992 by Massachusetts Institute of Technology
+ * Copyright FreeWnn Project.  2015
  *
  * Author: OMRON SOFTWARE Co., Ltd. <freewnn@rd.kyoto.omronsoft.co.jp>
+ * Maintainer:  FreeWnn Project
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -451,7 +453,7 @@ extern void print_out3 ();
 extern void print_out2 ();
 extern void print_out1 ();
 extern void print_out ();
-extern void print_msg_getc ();
+extern void print_msg_getc (register char *format, ...);
 extern void print_msg_wait ();
 
 /*
diff --git a/Xwnmo/xwnmo/printf.c b/Xwnmo/xwnmo/printf.c
index e90f9c6..4d0d529 100644
--- a/Xwnmo/xwnmo/printf.c
+++ b/Xwnmo/xwnmo/printf.c
@@ -11,8 +11,10 @@
  * Copyright OMRON Corporation. 1987, 1988, 1989, 1990, 1991, 1992, 1999
  * Copyright ASTEC, Inc. 1987, 1988, 1989, 1990, 1991, 1992
  * Copyright 1991, 1992 by Massachusetts Institute of Technology
+ * Copyright FreeWnn Project.  2015
  *
  * Author: OMRON SOFTWARE Co., Ltd. <freewnn@rd.kyoto.omronsoft.co.jp>
+ * Maintainer:  FreeWnn Project
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -41,6 +43,7 @@
         xwnmo
  */
 #include <stdio.h>
+#include <stdarg.h> /** add */
 #include "commonhd.h"
 #include "sdefine.h"
 #ifdef  XJUTIL
@@ -174,14 +177,18 @@ putchar_norm (c)
   pop_hrus ();
 }
 
+/** 2015-06-20: changed to ANSI style.  And used vsnprintf().  */
 void
-print_msg_getc (format, arg1, arg2, arg3)
-     register char *format;
-     char *arg1, *arg2, *arg3;
+print_msg_getc (register char *format, ...)
 {
   int ret;
 
-  sprintf (buf, format, arg1, arg2, arg3);
+  va_list ap;
+
+  va_start(ap, format);
+  vsnprintf(buf, sizeof buf, format, ap);
+  va_end(ap);
+  
   init_yes_or_no ((unsigned char *) buf, YesMessage);
   for (;;)
     {
